src_python/fl_client.py

```diff
@@ -0,0 +1,223 @@
+                    f1_test = "undefined"
+                    
+                logging.info('_____________________________________________________ Final model evalution ____________________________________________________________')
+                logging.info('Finel model (v%s) fetched',self.rounds)
+                logging.info('Training set : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s', eval[0][0], eval[0][1],eval[0][2],eval[0][3],f1_train,eval[0][4])
+                logging.info('Testing set : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',  eval[1][0], eval[1][1],eval[1][2],eval[1][3],f1_test,eval[1][4])
+                
+            else:
+                
+                self.rounds += 1
+                logging.info('_____________________________________________________ Training Round %s ____________________________________________________________',self.rounds)
+                logging.info('Global model v%s fetched',self.rounds - 1)
+
+                eval = self.MODEL.evaluate()
+
+                try:
+                    f1_train = (2 * eval[0][2] * eval[0][4]) / (eval[0][2] + eval[0][4])
+                    f1_test = (2 * eval[1][2] * eval[1][4]) / (eval[1][2] + eval[1][4])
+                except ZeroDivisionError as e:
+                    f1_train = "undefined"
+                    f1_test = "undefined"
+
+                logging.info('Global model v%s - Training set evaluation : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',self.rounds - 1, eval[0][0], eval[0][1],eval[0][2],eval[0][3],f1_train,eval[0][4])
+                logging.info('Global model v%s - Testing set evaluation : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',self.rounds - 1,  eval[1][0], eval[1][1],eval[1][2],eval[1][3],f1_test,eval[1][4])
+
+                
+                logging.info('Training started')
+                self.train()
+                logging.info('Training done')
+
+                # eval = self.MODEL.evaluate()
+
+                # f1_train = (2 * eval[0][2] * eval[0][4]) / (eval[0][2] + eval[0][4])
+                # f1_test = (2 * eval[1][2] * eval[1][4]) / (eval[1][2] + eval[1][4])
+                # logging.info('After Round %s - Local model - Training set evaluation : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',self.rounds, eval[0][0], eval[0][1],eval[0][2],eval[0][3],f1_train,eval[0][4])
+                # logging.info('After Round %s - Local model - Testing set evaluation : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',self.rounds, eval[1][0], eval[1][1],eval[1][2],eval[1][3],f1_test,eval[1][4])
+
+                logging.info('Sent local model to the server')
+                self.send_model()
+
+
+if __name__ == "__main__":
+
+    from models.supervised import Model
+
+    if 'IP' not in args.keys()  or args['IP'] == 'localhost':
```

Comment:
How do we set the IP and the PORT values?

Reply:
They will be passed as arguments when the script is called

Reply:
Ok